File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.java
class Solution {
	public int solution(int n, int a, int b) {
		int answer = 1;
		
		int big = a > b ? a : b;
		int small = a > b ? b : a;
		
		
		while(big - small != 1 || big - small == 1 && (big + small) % 4 != 3) {
			big = big % 2 == 0 ? big / 2 : big / 2 + 1;
			small = small % 2 == 0 ? small / 2 : small / 2 + 1;
			answer++;
		}
		
		return answer;
	}
}

File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.py
import math

def solution(n,a,b):
    answer = 1

    round_a = math.ceil(a / 2)
    round_b = math.ceil(b / 2)

    while round_a != round_b:
        round_a = math.ceil(round_a / 2)
        round_b = math.ceil(round_b / 2)

        answer += 1

    return answer
